python/tskit/trees.py

```diff
@@ -357,6 +357,19 @@ def __eq__(self, other):
             and self.metadata == other.metadata
         )
 
+    def alleles(self):
+        """
+        Return a set of all the alleles defined at this site (the ancestral state and
+        the derived states of any mutations).
+
+        .. note::
+            Unlike the .alleles attribute of the :class:`Variant` class, this does not
```

Comment:
I'm a bit fuzzy on whether there can also be situations where there would be alleles in this set that are not in the alleles returned by the variant.

I think it's important to stress the "this is a set" thing here too.

Reply:
> I'm a bit fuzzy on whether there can also be situations where there would be alleles in this set that are not in the alleles returned by the variant.

I think probably not: it looks like the alleles returned in the variants iterator include all the mutations in the TS, even if they are above nodes that are not in the tree topology:

```
import tskit
tables = tskit.TableCollection(1)
tables.nodes.add_row(0, 0)
tables.nodes.add_row(0, 0)
tables.sites.add_row(0, ancestral_state="A")
tables.mutations.add_row(site=0, derived_state="T", node=0)
tables.mutations.add_row(site=0, derived_state="G", node=1)
ts = tables.tree_sequence()
print("Num samples =", ts.num_samples, ". Alleles", next(ts.variants()).alleles)
```

> I think it's important to stress the "this is a set" thing here too.

Good point

Reply:
We now note: 

> This deliberately returns an (unordered) *set* of the possible allelic
            states (as defined by the site's ancestral allele and its associated
            mutations). If you wish to obtain an (ordered) *list* of alleles, for
            example to translate the numeric genotypes at a site into allelic states,
            you should instead use ``.alleles`` attribute of the :class:`Variant` class,
            which unlike this attribute includes ``None`` as a state when there is
            missing data at a site.